forge-gui/res/cardsfolder/upcoming/dreams_of_steel_and_oil.txt

```diff
@@ -0,0 +1,9 @@
+Name:Dreams of Steel and Oil
+ManaCost:B
+Types:Sorcery
+A:SP$ RevealHand | ValidTgts$ Opponent | TgtPrompt$ Select target opponent |ImprintRevealed$ True | RememberTargets$ True | SubAbility$ DBChooseCard | SpellDescription$ Target opponent reveals their hand. You choose an artifact or creature card from it, then choose an artifact or creature card from their graveyard. Exile the chosen cards.
+SVar:DBChooseCard:DB$ ChooseCard | ChoiceZone$ Hand | Choices$ Artifact.IsImprinted,Creature.IsImprinted | ChoiceTitle$ Choose an artifact or creature card | Mandatory$ True | SubAbility$ DBChooseFromGraveyard
+SVar:DBChooseFromGraveyard:DB$ ChooseCard | ChoiceZone$ Graveyard | Choices$ Artifact.RememberedPlayerOwn,Creature.RememberedPlayerOwn | ChoiceTitle$ Choose an artifact or creature card | Mandatory$ True | SubAbility$ DBExileBoth
+SVar:DBExileBoth:DB$ ChangeZone | Defined$ ChosenCard | Origin$ Graveyard | Origin$ Hand | Destination$ Exile | SubAbility$ DBCleanup
```

Comment:
This feels wrong... line 6 will overwrite ChosenCard from line 5, so this should only exile the card from line 6

I've been thinking about doing some kind of AdditiveChoice for ChooseCardEffect

For now I think you'll need to RememberChosen in line 5 and 6 - for this line you can use Defined$ RememberedCard

Reply:
Yeah should have mentioned that this card still needs some work. These cards with multiple targets in different zones are always a headache

Reply:
Hmm why does line 6 overwrite line 5. How can you do two successive choices ?

Reply:
ChooseCardEffect is currently only set up to SET chosen cards... which wipes any previously chosen cards

10 years ago when ChooseCardEffect was written it was all that was needed... like I said, there have been a few cards recently that have made me want to add an option.

Reply:
Ah thanks for the explanation ! I guess I'll just delete this for now.

Reply:
Or I guess a pump effect could work as a bandaid